### DISCUSSION TITLE
WireGuard transparent mode TLS handshake fails on Android

### DISCUSSION THREAD

**Discussion by cikabo** [mitmproxy/mitmproxy#6423] [Intercept Everything!]
Hi,
I followed the instructions to use the WireGuard transparent mode which I was expecting to overcome the certificate trust/pinning issue.

The web sites load in browser with no problems and the traffic is visible in the mitmproxy console and web interface.

However, many services fail to load with the TLS handshake error.

Example event log message:

[16:40:56.515][10.0.0.1:55176] client connect
[16:40:56.535][10.0.0.1:55176] server connect 172.217.16.170:443
[16:40:56.615][10.0.0.1:55176] Client TLS handshake failed. The client does not trust the proxy's certificate for play.googleapis.com (OpenSSL Error([('SSL routines', '', 'sslv3 alert certificate unknown')]))
[16:40:56.616][10.0.0.1:55176] client disconnect
[16:40:56.617][10.0.0.1:55176] server disconnect 172.217.16.170:443

Hopefully you can offer some insight...

Samsung Galaxy S22
Android version 13
One UI versioon 5.1

WireGuard for Android
v1.0.20231018

mitmproxy --version
Mitmproxy: 10.0.0
Python:    3.11.6
OpenSSL:   OpenSSL 3.1.3 19 Sep 2023
Platform:  macOS-13.6.1-x86_64-i386-64bit

**Comment by mhils** [comment]
WireGuard Mode solves the problem of getting traffic to mitmproxy, it does not help with/affect certificate trust/. certificate pinning.

**Reply by cikabo** [reply]
How could it be overcome apart from compiling the apps again?
I have installed the mitmproxy certificate as instructed... And the same setup works on iOS with no problems.
Thank you!

**Reply by mhils** [reply]
See https://docs.mitmproxy.org/stable/concepts-certificates/#certificate-pinning

**Reply by cikabo** [reply]
Just for me to make sure (sorry) - if I don't want to ignore the host/domain, and I can't change the app code or APK, I won't be able to see the traffic?
What about downloading the certificates and specifying using `--certs` option?
